### ISSUE TITLE
Order rules by dependencies

### ISSUE THREAD

**Issue by nykolaslima** [six2six/fixture-factory#72]
Today we have the following problem:
```
Fixture.of(User.class).addTemplate("valid", new Rule() {{
add("email", "${name}@gmail.com");
add("name", random("nykolas", "otavio"));
}};
```

The above code fails because email depends on `name` but when `email` is evaluated, `name` was not evaluated yet.

@Tavio has solved this problem on Angular-Fixture-Factory(https://github.com/Tavio/angular-fixture-factory) project using Topological sort(http://en.wikipedia.org/wiki/Topological_sorting)
https://github.com/Tavio/angular-fixture-factory/blob/master/angular-fixture-factory.js#L264-L312

We can use the same approach here.

**Comment by Tavio** [comment]
"Beware of bugs in the above code; I have only proved it correct, not tried it." - Knuth, Donald.

**Comment by ddomingues** [comment]
This issue shouldn't be closed? @nykolaslima 

And could you generate a new version with the PR #77?

**Comment by mahmoudimus** [comment]
This is indeed already merged via #77

**Comment by nykolaslima** [comment]
Thank you guys.
